.github/actions/e2e-testing/edge/action.yml

```diff
@@ -27,27 +31,56 @@ runs:
       uses: GuillaumeFalourd/clone-github-repo-action@main
       with:
         owner: 'pinecone-io'
-        repository: 'pinecone-rag-demo'
+        repository: 'ts-client-e2e-tests'
         branch: 'main'
 
-    - name: Move package to pinecone-rag-demo
-      run: cd pinecone-rag-demo && mv ../pinecone-database-pinecone-*.tgz .
+    - name: Move package to ts-client-e2e-tests
+      run: cd ts-client-e2e-tests && mv ../pinecone-database-pinecone-*.tgz .
       shell: bash
 
-    - name: Install rag-demo deps
-      run: cd pinecone-rag-demo && npm install
+    - name: Install e2e tests dependencies
+      run: cd ts-client-e2e-tests && npm install
       shell: bash
 
     - name: Install pinecone-ts-client "main" branch code into the Vercel app
-      run: cd pinecone-rag-demo && npm install pinecone-database-pinecone-*.tgz
+      run: cd ts-client-e2e-tests && npm install pinecone-database-pinecone-*.tgz
       shell: bash
 
     - name: Install Vercel CLI
-      run: cd pinecone-rag-demo && npm install vercel
+      run: cd ts-client-e2e-tests && npm install vercel
+      shell: bash
+
+    - name: Send POST request to endpoint Vercel app endpoint api/createSeedQuery
+      run: |
+        npm install -g typescript
+        npm install -g ts-node
+
+        API_KEY="${{ inputs.PINECONE_API_KEY }}"
+             
+        indexName=$(ts-node src/end-to-end/localRuns.ts $API_KEY)
+             
+        if [[ "$indexName" != *"edge-test"* ]]; then
+          echo "Error encountered: $indexName"
+          exit 1
+        else
+          echo "Test succeeded! Index created: $indexName"
+        fi
+
+        echo "indexName=$indexName" >> $GITHUB_ENV
```

Comment:
This seems fine for now, but it would be cool to eventually have some sort of reusable action that let you do something like this:

```yaml
- name: Call Vercel test endpoints
  uses: ./actions/post-vercel-endpoint
  with:
    app_url: https://ts-client-e2e-tests.vercel.app
    vercel_token: '${{ secrets.vercel_token }}'
    endpoints: 
      - api/createSeedQuery
      - api/test2
      - api/test3
```

In particular it seems like the ability to parameterize on app_url would be useful, since right now we can only deploy and test one branch at a time.

Reply:
cool!